### DISCUSSION TITLE
SuspendWorkflow question

### DISCUSSION THREAD

**Discussion by Aiol-CN** [danielgerlag/workflow-core#929] [Q&A]
a workflow has step: a,b,c & d;
at first, get workflowId by  StartWorkflow()  such as  
...
_workflowId = await _host.StartWorkflow<OrderData>(_workflowName, data); 
//_workflowId  value is "03549b82-4cc7-43dd-b927-de2e1ae75711"
...

and the workflow keeps running , then using SuspendWorkflow() when it was doing step b.
in my opinion, .SuspendWorkflow() can suspend the running workflow.so the step c & d will be executed after  ResumeWorkflow() down.

...
  var res = _workflowHost.SuspendWorkflow(_workflowId).Result;
// res value is false
...

but the real operation is the workflow running all steps and SuspendWorkflow() return false always.
what`s wrong?

**Comment by AngrySKL** [comment]
I downladed the source code and debug into it then found out that if fails to get the lock when it tries to suspend / resume, the lock seems to happen when step starts and release when step ends, still not get the idea how to make it really suspend

**Comment by q913777031** [comment]
I'm also having a problem with the glance, suspend the workflow is not effective
